### PR TITLE
[scripts][common-items]Add gelapod to trash bins

### PR DESCRIPTION
**Pull request by Raykyn55**

### CODE REVIEW

**Comment by Raykyn55** [comment]
I believe theres a different command for lower level characters, but I'm not sure if it changes the beginning of the message or the end. my recollection is that it is at the end, something about almost losing fingers.

**Comment by Raykyn55** [comment]
```
[pick]>open my mahogany chest
You open the mahogany chest...

In the mahogany chest you see some nemoih root, a shining calavarite runestone, a medium grey diamond, and a large smoky clear diamond.
s>
[pick]>fill my green pouch with my mahogany chest
Gained: Locksmithing(+2)
You open your green gem pouch and quickly fill it with 2 gems from your mahogany chest, closing it quickly.
s>
[pick]>look in my mahogany chest
In the mahogany chest you see some nemoih root and a shining calavarite runestone.
s>
[pick]>get root from my mahogany chest
s>
You get some nemoih root from inside your mahogany chest.
s>
[pick]>feed my root to gelapod
You toss some nemoih root deftly at the domesticated gelapod which folds its mouth around it eagerly, then sidles up next to you, as if waiting for more.
s>
[pick]>get runestone from my mahogany chest
You get a shining calavarite runestone from inside your mahogany chest.
s>
[pick]>feed my runestone to gelapod
s>
You toss a shining calavarite runestone deftly at the domesticated gelapod which folds its mouth around it eagerly, then sidles up next to you, as if waiting for more.
s>
[pick]>dismantle my mahogany chest thump
With a smirk of supreme confidence, you rap the mahogany chest once with a knuckle.  The chest collapses and its pieces fall to the ground in what can only be described as a surreally ordered fashion.  You could almost reassemble the neatly laid out pieces, needing no plans for its construction.  Order being something of an anathema to you, you kick the pieces aside.
Roundtime: 3 seconds.
```

**Comment by MahtraDR** [comment]
How do I test this?  How do I get a gelapod?

I'm not sure I want that change in drinfomon. What's the purpose of that change?

**Comment by Raykyn55** [comment]
> How do I test this? How do I get a gelapod?
> 
> I'm not sure I want that change in drinfomon. What's the purpose of that change?

It's the trash npc at the pond in xing.

**Comment by Raykyn55** [comment]
That little method in drinfomon is a filter that removes npcs from the list of room_objs, turning:
```
>l i
You take a moment to look for all the items in the area and see a town guard, a young alchemist student, the Amphitheater gate, a gilded walkway, a grass path leading behind the amphitheater, a strong and stately mature oak and a waste bin.
```
into:
```
>;e echo DRRoom.room_objs
--- Lich: exec1 active.
[exec1: ["strong", "stately mature oak", "gilded walkway", "the Amphitheater gate", "grass path leading behind the amphitheater", "waste bin"]]
--- Lich: exec1 has exited.
```
Note the guard and alchemist are not present in the list. With this change, if and only if the object is the domesticated gelapod, we add that to the list to be accessed by dispose_trash method (near the top):
```ruby
    trashcans = DRRoom.room_objs
                      .reject { |obj| obj =~ /azure \w+ tree/ }
                      .map { |long_name| DRC.get_noun(long_name) }
                      .select { |obj| @@trash_storage.include?(obj) }
```
which allows:
```
>;e echo DRCI.dispose_trash(DRC.right_hand)
--- Lich: exec1 active.
[exec1]>feed my elbaite runestone to gelapod
You toss a colorful elbaite runestone deftly at the domesticated gelapod which folds its mouth around it eagerly, then sidles up next to you, as if waiting for more.
>
[exec1: true]
--- Lich: exec1 has exited.

File: drinfomon.lic [hunk 1]
```diff
@@ -609,8 +609,8 @@ end
 
 def find_objects(room_objs)
   clean_and_split(room_objs)
-    .reject { |obj| obj =~ /pushBold/ }
-    .map { |obj| obj.sub(/\.$/, '').strip.sub(/^a /, '').strip.sub(/^some /, '') }
+    .reject { |obj| obj =~ /pushBold/ unless obj =~ /domesticated gelapod/ }
+    .map { |obj| obj.sub(/\.$/, '').sub(/<pushBold\/>/, '').sub(/<popBold\/>/, '').strip.sub(/^a /, '').strip.sub(/^some /, '') }
```
Comment (MahtraDR) [inline drinfomon.lic]:
```suggestion
```

File: common-items.lic [hunk 2]
```diff
@@ -397,9 +401,13 @@ module DRCI
         trashcan = 'dark gap' if DRRoom.room_objs.include?('tangle of thick roots forming a dark gap')
       elsif XMLData.room_title == '[[A Junk Yard]]'
         trashcan = 'bin'
+      elsif trashcan == 'gelapod'
+        trash_command = "feed my #{item} to gelapod"
       end
+      
+      trash_command ||= "put my #{item} in #{trashcan}"
```
Comment (MahtraDR) [inline common-items.lic:408]:
```suggestion
      trash_command = "put my #{item} in #{trashcan}" unless trashcan == 'gelapod'
```

**Comment by MahtraDR** [comment]
Also updated https://github.com/rpherbig/dr-scripts/wiki/Adding-new-trash-receptacles to account for this PR, and https://github.com/rpherbig/dr-scripts/pull/6412

**Pull request merged**